src/DeviceDataStreamValues.tsx

```diff
@@ -0,0 +1,312 @@
+  aggregation: AstarteInterface['aggregation'];
+}
+
+const InterfaceData = ({ interfaceData, aggregation }: InterfaceDataProps): React.ReactElement => {
+  if (interfaceData.length === 0) {
+    return <p>No data in the selected timeframe.</p>;
+  } else if (aggregation === 'individual') {
+    return <IndividualDatastreamTable data={interfaceData as AstarteDatastreamIndividualData[]} />;
+  }
+  return <ObjectDatastreamTable data={interfaceData as AstarteDatastreamObjectData[]} />;
+};
+
+export default (): React.ReactElement => {
+  const { interfaceName = '', deviceId = '', interfaceMajor } = useParams();
+  const astarte = useAstarte();
+  const [requestingData, setRequestingData] = useState(false);
+  const [fetchedData, setFetchedData] = useState<AstarteInterfaceValues | null>(null);
+  const [formAlerts, formAlertsController] = useAlerts();
+  const [interfaceData, setInterfaceData] = useState<AstarteInterface | null>(null);
+  const [interfacePaths, setInterfacePaths] = useState<string[]>([]);
+  const [selectedPath, setSelectedPath] = useState<string>('');
+
+  const fetchData = async (path: string, since?: string, to?: string) => {
+    setRequestingData(true);
+    setFetchedData(null);
+    astarte.client
+      .getDeviceData({
+        deviceId,
+        interfaceName,
+        path,
+        since,
+        to,
+      })
+      .then((data) => {
+        setSelectedPath(path);
+        setFetchedData(data);
+      })
+      .catch((err) => {
+        formAlertsController.showError(
+          `Could not fetch data to interface: ${err.response.data.errors.detail}`,
+        );
+        setFetchedData([]);
+      })
+      .finally(() => {
+        setRequestingData(false);
+      });
+  };
+
+  const transformedData =
+    interfaceData && fetchedData
+      ? interfaceData.aggregation === 'individual'
+        ? transformIndividualDatastreamInterfaceValues(selectedPath, fetchedData)
+        : transformAggregatedDatastreamInterfaceValues(selectedPath, fetchedData)
+      : [];
+
+  useEffect(() => {
+    if (interfaceMajor) {
+      const major = parseInt(interfaceMajor, 10);
+      astarte.client
+        .getInterface({ interfaceName, interfaceMajor: major })
+        .then((interfaceData) => {
+          setInterfaceData(interfaceData);
+          astarte.client
+            .getDeviceData({ deviceId: deviceId, interfaceName: interfaceName })
+            .then((deviceData) => {
+              const placeholder = Object.keys(deviceData)[0];
+              const listOfExistingEndpoint = interfaceData.mappings.map((data) => data.endpoint);
+              let modifiedEndpoints: string[];
+              if (interfaceData.aggregation === 'object') {
+                modifiedEndpoints = [`/${placeholder}`];
+              } else {
+                modifiedEndpoints = replacePlaceholders(listOfExistingEndpoint, placeholder);
+              }
+
+              setInterfacePaths(modifiedEndpoints);
+            });
```

Comment:
I suppose you're doing this because you may have a parametric endpoint in the Astarte interface and you need to know what params were published.
However:
- it seems that only the first param is considered: `const placeholder = Object.keys(deviceData)[0]`
- the param maybe have different locations in the endpoint, we cannot assume it is the first path token: `/%{param}/endpoint`, `/another/%{param}/endpoint`, `/yet/another/%{param}/endpoint`.